picamera2/request.py

```diff
@@ -203,7 +203,11 @@ def make_array(self, buffer, config):
             # cv2.cvtColor(image, cv2.COLOR_YUV2BGR_YUYV) will convert directly to RGB.
             image = array.reshape(h, stride // 2, 2)
         elif fmt == "MJPEG":
-            image = np.array(Image.open(io.BytesIO(array)))
+            image = Image.open(io.BytesIO(array))
+            if drm:
+                image = image.convert('RGBA').tobytes()
+            else:
+                image = np.array(img)
```

Comment:
With those other tweaks I think we can revert this change, and then also remove the extra `drm` parameter, leaving things a bit simpler.